internal.nix

```diff
@@ -133,7 +188,38 @@ rec {
       passthru.node_modules = nm;
     };
 
-  build = attrs:
-    let nm = node_modules attrs; in
-    { };
+  build =
+    { src
+    , npmCommands
+    , installPhase
+    , symlink_node_modules ? true
+    , copy_node_modules ? false
+    , buildBuildInputs ? [ ]
+    , ...
+    }@attrs:
+    let
+      nm = node_modules (builtins.removeAttrs attrs [ "buildBuildInputs" "copy_node_modules" "npmCommands" "installPhase" ]);
+    in
+    stdenv.mkDerivation {
+      pname = nm.pname;
+      version = nm.version;
+      buildInputs = [ nm ] ++ buildBuildInputs;
+
+      postUnpack =
+        if !copy_node_modules && symlink_node_modules then ''
+          ln -sf ${nm}/node_modules node_modules
+          export NODE_PATH="$(pwd)/node_modules:$NODE_PATH"
```

Comment:
This is a bit duplicated and the copy vs the symlink code is really unfortunate. I guess we'll run into a bunch of those. 

A single flag like `node_modules_mode` (or better) that can either be `"symlink"` or `"copy"` would be nice.

Reply:
Added said enum with shared code for both shell and build expression. Also a few tests cover these flags now.